test/firebase_auth_mocks_test.dart

```diff
@@ -99,6 +101,23 @@ void main() {
       expect(auth.userChanges(), emitsInOrder([null, isA<User>()]));
       expect(user.isAnonymous, isTrue);
     });
+
+    test('should send sms code to user', () async {
+      final auth = MockFirebaseAuth();
+      final codeSent = Completer<bool>();
+      // currently does not auto retrieve the sms code like it does on Android
+      // it is left to the user to use the code to sign in, so we encompass
+      // every platform
```

Comment:
Thanks for the example and comment. Reading https://firebase.flutter.dev/docs/auth/phone/#codesent was helpful too.